### ISSUE TITLE
Console buffer after interactive rebase

### ISSUE THREAD

**Issue by marc-nijland** [git-for-windows/git#2750]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

git version 2.27.0.windows.1

```
$ git --version --build-options

git version 2.27.0.windows.1
cpu: x86_64
built from commit: 907ab1011dce9112700498e034b974ba60f8b407
sizeof-long: 4
sizeof-size_t: 8

```

Windows 10

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.18363.900]
```
Default options
```
> type "C:\Program Files\Git\etc\install-options.txt"

Editor Option: VIM
Custom Editor Path:
Path Option: CmdTools
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: ConHost
Git Pull Behavior Option: Rebase
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git rebase -i <commithash>
```
 - What did you expect to occur after running these commands?

The cursor on the next line

 - What actually happened instead?

Console buffer is from the last command til the end of the window (not the buffer) filled with empty lines and the cursor is at the bottom. The interactive editor is vim (in git\usr\bin) and vim does not have that problem after creating a new file.
EDIT: The buffer from vim is not erased and remains in the console buffer, plus extra blank lines.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Sorry, this is not public

**Comment by dscho** [comment]
Hmm. Could you try again after enabling support for the Pseudo Console?

**Comment by marc-nijland** [comment]
I just tested on 2.28 and fortunately it doesn't work better. There seems to be a slight difference: the console is scrolled down and on the first line I now see _Successfully rebased and updated refs/heads/..._ But when I scroll up there still appears the buffer from the rebase interface.

**Comment by dscho** [comment]
@marc-nijland just to be sure, did you enable Pseudo Console support?

**Comment by marc-nijland** [comment]
Of course I did :)

**Comment by dscho** [comment]
@marc-nijland I had to ask because it was not spelled out. And the original report said the opposite:

> Enable Pseudo Console Support: Disabled

Now, I am not sure that I really understand what the exact symptom is. Would you mind showing a screenshot of the issue? I will then try to reproduce (but I need to caution against high expectations: the Pseudo Console patches were quite intrusive and introduced even regressions in the mode where Pseudo Console support is turned off, therefore I fully expect this to be hard to debug and to fix, and I do not have the time right now, nor do I know anybody with enough expertise who would have that time right now).

**Comment by stanio** [comment]
Not sure if I'm seeing the same.  I'm seeing this pretty much since the first version mentioning support for pseudo consoles, but if I remember correctly I've seen it with both pseudo console support enabled and disabled:
```
$ git rebase -i master
Created autostash: 2a0d7f5a6
[detached HEAD 2ca5f4ad3] Hidpi, Preferences: Scale icons with text option
                                                                           Date: Fri Oct 2 18:31:44 2020 +0300
                                                                                                               4 files changed, 28 insertions(+), 1 deletion(-)
[detached HEAD f14637d67] Show selected alias URL in status bar
                                                                Date: Sun Jun 13 13:18:41 2021 +0300
                                                                                                     5 files changed, 95 insertions(+), 4 deletions(-)
Applied autostash.Rebasing (23/35)
Successfully rebased and updated refs/heads/work-in-progress.
```
I've selected to `reword` two of the commits (without actually changing the message).  Whenever rebase stops for editing or resolving conflicts the output appears like that – it seems there are LFs that don't additionally cause CR, so the text continues on the next line at the last position from the previous line.

**Comment by dscho** [comment]
I suspect that this bug is fixed in v2.34.0.

**Comment by stanio** [comment]
I confirm the issue with the new lines I've described is fixed in v2.34.0 for me.

**Comment by uniphiny** [comment]
uni

**Comment by dscho** [comment]
> I confirm the issue with the new lines I've described is fixed in v2.34.0 for me.

Awesome!
